### PR TITLE
Support for Android armv7 and aarch64 target triples

### PR DESCRIPTION
**Pull request by MortimerGoro**
See https://github.com/servo/servo/issues/11921 and https://github.com/servo/servo/pull/16769

---
This change is [Reviewable](https://reviewable.io/reviews/servo/mozjs/118)

### CODE REVIEW

File: makefile.cargo [hunk 1]
```diff
@@ -11,26 +11,30 @@ CONFIGURE_FLAGS := \
...
 ifneq ($(HOST),$(TARGET))
 
-CC ?= $(TARGET)-gcc
-CPP ?= $(TARGET)-gcc -E
-CXX ?= $(TARGET)-g++
-AR ?= $(TARGET)-ar
-
-CONFIGURE_FLAGS += --target=$(TARGET) --disable-gold
+	ifeq (armv7-linux-androideabi,$(TARGET))
+		# Fix: armv7 TARGET used by rust is not the same as the TARGET needed for CXX toolchain.
```
Comment (jdm) [inline makefile.cargo]: What does the `Fix:` prefix here mean? I first interpreted it as FIXME, but I don't think that's correct.

File: mozjs/build/moz.configure/rust.configure [hunk 2]
```diff
@@ -92,6 +92,8 @@ def rust_target(rust_compiler, rustc, target, cross_compiling):
             # Android
             ('x86', 'Android'): 'i686-linux-android',
             ('arm', 'Android'): 'arm-linux-androideabi',
+            ('aarch64', 'Android'): 'aarch64-linux-android',
+            ('armv7', 'Android'): 'arm-linux-androideabi',
```
Comment (jdm) [inline mozjs/build/moz.configure/rust.configure]: The other changes in this PR will need to be [submitted upstream](https://bugzilla.mozilla.org/enter_bug.cgi?product=Core&component=JavaScript%20Engine) if they're not [already present](https://hg.mozilla.org/mozilla-central/file/38f862749aed/js).

File: makefile.cargo (hunk 1 above)
Comment (MortimerGoro) [inline makefile.cargo]: It's a typo, I'll fix it

File: mozjs/build/moz.configure/rust.configure (hunk 2 above)
Comment (MortimerGoro) [inline mozjs/build/moz.configure/rust.configure]: So the way to get those changes in this repo is first waiting for a merge in upstream and then running this? https://github.com/servo/mozjs#upgrading

Comment (jdm) [inline mozjs/build/moz.configure/rust.configure]: No, it's fine to merge them in here. I just want to make sure that they don't get lost as part of the next upgrade we do (which has no definite timeline, unfortunately).

**Comment by MortimerGoro** [comment]
Fixed the typo. I'll submit the required changes upstream.

**Comment by jdm** [comment]
Looks good.
@bors-servo: r+

**Comment by larsbergstrom** [comment]
@MortimerGoro If you need a point of contact on the Firefox/Gecko side for help with Android Rust-related build changes, @froydnj may be able to help :-)

**Pull request merged**